src/taskrunner.py

```diff
@@ -0,0 +1,241 @@
+
+    if escaped_sep not in argstr:
+        return argstr.split(sep)
+
+    before, _, after = argstr.partition(escaped_sep)
+    startlist = before.split(sep)  # a regular split is fine here
+    unfinished = startlist[-1]
+    startlist = startlist[:-1]
+
+    # recurse because there may be more escaped separators
+    endlist = _escape_split(sep, after)
+
+    # finish building the escaped value. we use endlist[0] becaue the first
+    # part of the string sent in recursion is the rest of the escaped value.
+    unfinished += sep + endlist[0]
+
+    return startlist + [unfinished] + endlist[1:]  # put together all the parts
+
+def parse_arguments(arguments):
+    """
+    Parse string list into list of tuples: command, args, kwargs, hosts, roles.
+
+    See sites/docs/usage/fab.rst, section on "per-task arguments" for details.
+    """
+    cmds = []
+    for cmd in arguments:
+        args = []
+        kwargs = {}
+        if ':' in cmd:
+            cmd, argstr = cmd.split(':', 1)
+            for pair in _escape_split(',', argstr):
+                result = _escape_split('=', pair)
+                if len(result) > 1:
+                    k, v = result
+                    kwargs[k] = v
+                else:
+                    args.append(result[0])
+        cmds.append((cmd, args, kwargs))
+    return cmds
+
+# --- end
+
+def exec_task(task_str, task_map_list):
+
+    task_name, task_args, task_kwargs = parse_arguments([task_str])[0]
+
+    return_map = {
+        'task': task_name,
+        'task_args': task_args,
+        'task_kwargs': task_kwargs
+    }
+
+    task_map_list = [t for t in task_map_list if t['name'] == task_name]
+    if not task_map_list:
+        print("Command not found: %r" % task_name)
+        return_map['rc'] = 1
+        return return_map
+
+    try:
+        task_map = task_map_list[0]
+        return_map['result'] = task_map['fn'](*task_args, **task_kwargs)
+        return_map['rc'] = 0
+        return return_map
+
+    except KeyboardInterrupt:
+        print('\nStopped.') # mimic fabric
+        return_map['rc'] = 1
+        return return_map
+
+    except BaseException as e:
+        print('exception while executing task %r: %s\n' % (task_name, str(e)))
+        print(traceback.format_exc())
+        return_map['rc'] = 2 # arbitrary
+        return return_map
+
+def main(arg_list):
+    show_debug_tasks = os.environ.get("BLDR_ROLE") == "admin"
+    task_map_list = generate_task_list(show_debug_tasks)
+
+    if not arg_list or arg_list[1:]:
+        print("`taskrunner.main` must be called from the ./bldr script")
+        return 1
+
+    # bash hands us an escaped string value via ./bldr
+    command_string = arg_list[0].strip()
+
+    if not command_string or command_string in ["-l", "--list", "-h", "--help", "-?"]:
+        print("Available commands:\n")
+        indent = 4
+        for tm in task_map_list:
+            path_len = len(tm['name'])
+            max_path_len = 35
```

Comment:
arbitrary? Just for output, I guess so not important if that's the case

Reply:
it's the length of the longest task name ('path') in the list of tasks. you take an initial indent, plus that magic number, plus a dynamic offset so that the description strings that come next are aligned. I'll make sure this is a calculated value.